### DISCUSSION TITLE
Source distribution of 0.11.0 is missing the `docs_src` folder

### DISCUSSION THREAD

**Discussion by theMarix** [fastapi/typer#771] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
#!/bin/sh

pip download --no-binary :all: typer
tar -xaf typer-0.11.0.tar.gz
cd typer-0.11.0
pytest
```

### Description

As of 0.11.0, the source distribution of typer on PyPI no longer seems to contain the `docs_src` directory. As, however, all of the tutorial tests import from this directory, all of them fail. Just to give one example:
```
____________________________________________________________ ERROR collecting tests/test_completion/test_completion.py ____________________________________________________________
ImportError while importing test module '/home/marix/openSUSE_Build_Service/devel:languages:python/python-typer/tmp/typer-0.11.0/tests/test_completion/test_completion.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib64/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_completion/test_completion.py:6: in <module>
    from docs_src.commands.index import tutorial001 as mod
E   ModuleNotFoundError: No module named 'docs_src'
```
Admittedly running the tests from the source distribution is somewhat of a fringe case. But when packaging for distributions that's the easiest way.

### Operating System

Linux

### Operating System Details

This is on openSUSE but the problem should not be affected by that.

### Typer Version

0.11.0

### Python Version

3.11 but affects any.

**Comment by svlandeg** [comment, marked as answer]
Thanks for the report! 🙏 
We'll look into this.

**Reply by tiangolo** [reply]
Indeed! Most probably an oversight during the transition to PDM. I created an issue here: https://github.com/tiangolo/typer/issues/772, and I'll look into it.

I'm about to do some other related changes, there will be a `typer-slim` package with the same in current `typer`, and `typer` will just require `typer-slim[standard]`, so it will include by default `rich`, `shellingham`, etc.

I suspect this could also affect repackers (I'm still not sure why repackaging is necessary, but anyway :sweat_smile:).

I would like to have a test to make sure the repackaging works correctly, @theMarix would you like to contribute that? I imagine it would be something like using `build` to generate an sdist and then something like whatever you would do to test and simulate repackaging it.

**Reply by tiangolo** [reply]
This was handled in https://github.com/tiangolo/typer/pull/773, now the sdist includes the necessary files to install dependencies and test it before repackaging it. It will be available in Typer 0.11.1 in the next hours. :rocket:

**Reply by theMarix** [reply]
Hi @tiangolo . Thank you for the extremely quick fix. I see you also already added the test, so I assume your question has become obsolete. It's essentially the same I would have done.

The purpose of repackaging is solely to enable software that utilises other package management systems to rely on Typer. You obviously don't need that for developer environments where everyone uses virtual environments, or for server use case where everthing is in containers. But Desktop applications, despite the rise of Flatpack and Snap, still seldom are. And there is a benefit to being able to having the system updater being able to pull in dependency updates for all installed applications.

I really appreciate the support you provide! It would have been completely valid to just ask me to build directly from Git instead of satisfying my inner Monk by fixing the sdist to be in a consistent state again. :)

**Reply by tiangolo** [reply]
Thanks for the feedback @theMarix!

BTW, yesterday I released version `0.12.0` that works a bit differently, and I think you (and any others repackaging Typer) would be the only ones interested in the details :sweat_smile: 

So here they are:

* Now there's a package `typer-slim`, this is the equivalent of what `typer` has been, that's what exposes the Python module `typer` (as in `from typer import Typer`). This one has an extras `standard` that includes `rich` and `shellingham`. Before, the previous package `typer` had extras `all`, it is now renamed `standard`.
* There was a package called `typer-cli` that had some code to expose a `typer` command in the shell. That code was merged/included in this repo. And now there is a a new `typer-cli` package that doesn't contain any code, it only exposes the `typer` shell command (which is just the same as `python -m typer`).
* There is now a package `typer` that doesn't contain any code, just dependencies, it depends on `typer-slim[default]` and `typer-cli`. So, when people `pip install typer` they get the default `standard` dependencies and the `typer` command.

This setup is only to provide the standard dependencies by default in the simplest to install process (`pip install typer`) without requiring newbies to add extras, but to allow advanced users to opt-out of the extras if they want by installing `typer-slim` instead.

---

Now, for you that means that if you want to test the sdist, it now should be the same process but for `typer-slim`. And I imagine you would repack the new `typer` package, maybe `typer-slim` as well.

**Reply by tiangolo** [reply]
Another update, having `typer` depend on `typer-slim[standard]` was problematic because `pip` would install and copy the new `typer` directory and then *after* remove the "old" `typer` directory, which means it would remove most of the files. Because of that, I restructured it all, and now `typer` doesn't depend on `typer-slim`, it includes all the code and tests, and declares the default dependencies.

`typer-slim` is the same but without the default dependencies, they are in an optional dependencies group `standard`.

So, now you should be able to repack the same `typer` directly. Optional `typer-slim` if that's needed, but `typer` alone should work fine.

**Reply by theMarix** [reply]
Thanks for the hint. I had seen that already. But it's always better to notice twice than not at all. :)

What's a little unclear to me though: This does make `typer-slim` and `typer` mutually exclusive, does it not? Pip, I assume, does notice the conflict in file ownership but uninstalling either would break the other one in that case.
